scripts/versiondb/updateversiondb.jl

```diff
@@ -15,19 +15,12 @@ function remove_prefix(s, prefix)
 end
 
 function triplet2channel(triplet)
```

Comment:
Unrelated to this PR, why this doesn't use parsing in Base.BinaryPlatforms instead of reinventing the parsing (and doing it slightly wrong)?

Reply:
What is wrong about it?

Also, this is not a parsing function, it is a map from Julia platforms to Juliaup channels. Right now, it could be replaced by parsing, but in the past I used it (if I remember correctly) to shim for things like non-existing binaries on certain platforms. I think I used this (in some way) for example to handle the Mac M chips before we had native Julia builds for that.

Reply:
For example hardcoding the Darwin version in the check can be wrong.

Reply:
No, there is nothing wrong here. You are making assumptions here how that function is used that are not correct. At the end of the day this code will only ever run on a hand-curated list of platforms, and then we need to hand curate the mapping accordingly. That is on purpose, I want errors to be thrown if something about the platform lists changes and I forget to adjust this mapping.